src/devices/UFireIse/UFireIse.cs

```diff
@@ -0,0 +1,415 @@
+        public void SetDualPointCalibration(ElectricPotential refLow, ElectricPotential refHigh, ElectricPotential readLow, ElectricPotential readHigh)
+        {
+            if (refLow.Millivolts >= -1024 && refLow.Millivolts <= 1024 &&
+                refHigh.Millivolts >= -1024 && refHigh.Millivolts <= 1024 &&
+                readLow.Millivolts >= -1024 && readLow.Millivolts <= 1024 &&
+                readHigh.Millivolts >= -1024 && readHigh.Millivolts <= 1024)
+            {
+                WriteRegister(Register.ISE_CALIBRATE_REFLOW_REGISTER, Convert.ToSingle(refLow.Millivolts));
+                WriteRegister(Register.ISE_CALIBRATE_REFHIGH_REGISTER, Convert.ToSingle(refHigh));
+                WriteRegister(Register.ISE_CALIBRATE_READLOW_REGISTER, Convert.ToSingle(readLow));
+                WriteRegister(Register.ISE_CALIBRATE_READHIGH_REGISTER, Convert.ToSingle(readHigh));
+            }
+
+            throw new ArgumentOutOfRangeException("The value range is -1024 mV to 1024 mV for refLow, refHigh, readLow and readLow");
+        }
+
+        /// <summary>
+        /// Changes the default I2C address
+        /// </summary>
+        /// <param name="i2cAddress">The new I2C address. Range: 1 to 127 </param>
+        public void SetI2cAddressAndDispose(byte i2cAddress)
+        {
+            if (i2cAddress >= 1 && i2cAddress <= 127)
+            {
+                WriteRegister(Register.ISE_SOLUTION_REGISTER, Convert.ToByte(i2cAddress));
+                SendCommand((byte)Command.ISE_I2C);
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException("The value range is 1 to 127");
+            }
+
+            Dispose();
+        }
+
+        /// <summary>
+        /// Low level reads of EEPROM
+        /// </summary>
+        /// <param name="address">Address to read</param>
+        /// <returns>Low level data from EEPROM</returns>
+        public float ReadEeprom(Register address)
+        {
+            WriteRegister(Register.ISE_SOLUTION_REGISTER, (byte)address);
+            SendCommand((byte)Command.ISE_MEMORY_READ);
+
+            return ReadFloat(Register.ISE_BUFFER_REGISTER);
+        }
+
+        /// <summary>
+        /// Low level write of EEPROM
+        /// </summary>
+        /// <param name="address">Address</param>
+        /// <param name="data">Data to write to the address</param>
+        public void WriteEeprom(Register address, float data)
+        {
+            WriteRegister(Register.ISE_SOLUTION_REGISTER, (byte)address);
+            WriteRegister(Register.ISE_BUFFER_REGISTER, data);
+            SendCommand((byte)Command.ISE_MEMORY_WRITE);
+        }
+
+        /// <summary>
+        /// Get Firmware version
+        /// </summary>
+        /// <returns>The version of the firmware</returns>
+        public Version GetFirmwareVersion()
+        {
+            return new Version(0, (int)ReadByte(Register.ISE_FW_VERSION_REGISTER));
```

Comment:
should you swap arguments here? in the other Version you've read float, now you read byte (this one seems more correct). Likely you don't need to cast byte to int as it should upgrade the type automatically

Reply:
I have changed it back to a string, because the manufacturer do not provide any information about the format of the version number. This insures it supports any format it sends back.

Reply:
uFire treats it as byte: https://github.com/u-fire/Isolated_ISE/blob/b2d58b40a7873cda98796f714272695bec365cea/python/RaspberryPi/uFire_ISE.py#L151 - change it to major or minor. regardless be consistent with other version API - I don't think float is correct